File: srm188/MagicSquare.java
import java.lang.*;

public class MagicSquare {

	public int missing(int[] square) {
		int mn = 0;
		int ind = 0;
		for (int i = 0; i < square.length; i++) {
			if (square[i] == -1) {
				ind = i;
			}
		}
		if (ind < 3) {
			mn = square[3] + square[4] + square[5] - square[0] - square[2] - square[1] - 1;
			} else if (ind >= 3 && ind < 6) {
				mn = square[0] + square[2] + square[1] - 1 - square[3] - square[4] - square[5];
			} else {
				mn = square[0] + square[2] + square[1] - 1 - square[6] - square[7] - square[8];
		}
	return mn;
	}
}

File: srm188/magic_square.py
class MagicSquare:
	def missing(self, square):
		mn = 0
		ind = square.index(-1)
		if ind < 3: mn = sum(square[3:6]) - sum(square[0:3]) - 1
		elif ind >= 3 and ind < 6: mn = sum(square[0:3]) - sum(square[3:6]) - 1  
		else: mn = sum(square[0:3]) - sum(square[6:]) - 1
		return mn